cpu/samd21/cpu.c

```diff
@@ -168,8 +195,8 @@ static void clk_init(void)
     while ((SYSCTRL->PCLKSR.reg & mask) != mask) { } /* Wait for DFLL lock */
 
     /* select the DFLL as source for clock generator 0 (CPU core clock) */
-    GCLK->GENDIV.reg =  (GCLK_GENDIV_DIV(1U) | GCLK_GENDIV_ID(0));
-    GCLK->GENCTRL.reg = (GCLK_GENCTRL_GENEN | GCLK_GENCTRL_SRC_DFLL48M | GCLK_GENCTRL_ID(0));
+    GCLK->GENDIV.reg =  (GCLK_GENDIV_DIV(1U) | GCLK_GENDIV_ID(SAM0_GCLK_MAIN));
+    GCLK->GENCTRL.reg = (GCLK_GENCTRL_GENEN | GCLK_GENCTRL_SRC_DFLL48M | GCLK_GENCTRL_ID(SAM0_GCLK_MAIN));
     GCLK->CLKCTRL.reg = GCLK_CLKCTRL_CLKEN | GCLK_CLKCTRL_GEN_GCLK0;
```

Comment:
Perhaps we should change this line as well, just to make everything consistent.
```
GCLK->CLKCTRL.reg = GCLK_CLKCTRL_CLKEN | GCLK_CLKCTRL_GEN_GCLK0;
```
=>
```
GCLK->CLKCTRL.reg = GCLK_CLKCTRL_GEN(SAM0_GCLK_MAIN) |
                                       GCLK_CLKCTRL_ID_DFLL48 |
                                       GCLK_CLKCTRL_CLKEN;
```

Reply:
Good idea!